src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -92,6 +97,32 @@ private void deleteFile(File fileToDelete) {
   }
 
   /**
+   * Create Symbolc Link for the HDFS binary.
+   **/
+  private void createSymbolicLink() {
+    log.info("Creating a symbolic link for HDFS binary");
+    try {
+      // Get hdfs executable direcotry
+      File sandboxHdfsBinary = new File(System.getProperty("user.dir"));
+      Path sandboxHdfsBinaryPath = Paths.get(sandboxHdfsBinary.getAbsolutePath());
+
+      // Delete and recreate directory for symbolic link every time
+      File hdfsBinaryDir = new File(HDFSConstants.HDFS_BINARY_PATH);
+      File baseDir = new File(HDFSConstants.MESOSPHERE_OPT_PATH);
+      if (hdfsBinaryDir.exists()) {
+        deleteFile(hdfsBinaryDir);
+      } else if (!baseDir.exists()) {
```

Comment:
Ah, because it's the parent directory. I agree it's probably less confusing as an if statement though so I will update it. thx